Tests fail when commit.gpgsign = true

```bash
$ git config --global commit.gpgsign true
$ cpan -i App::Git::Autofixup
CPAN: LWP::UserAgent loaded ok (v6.43)
Fetching with LWP:
http://www.cpan.org/authors/01mailrc.txt.gz
Reading '/home/john/.local/share/.cpan/sources/authors/01mailrc.txt.gz'
CPAN: Compress::Zlib loaded ok (v2.084)
............................................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/02packages.details.txt.gz
Reading '/home/john/.local/share/.cpan/sources/modules/02packages.details.txt.gz'
  Database was generated on Thu, 01 Apr 2021 16:29:02 GMT
.............
  New CPAN.pm version (v2.28) available.
  [Currently running version is v2.22]
  You might want to try
    install CPAN
    reload cpan
  to both upgrade CPAN.pm and run the new version without leaving
  the current session.


...............................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/03modlist.data.gz
Reading '/home/john/.local/share/.cpan/sources/modules/03modlist.data.gz'
DONE
Writing /home/john/.local/share/.cpan/Metadata
Running install for module 'App::Git::Autofixup'
Fetching with LWP:
http://www.cpan.org/authors/id/T/TO/TORBIAK/App-Git-Autofixup-0.003001.tar.gz
CPAN: Digest::SHA loaded ok (v6.02)
Fetching with LWP:
http://www.cpan.org/authors/id/T/TO/TORBIAK/CHECKSUMS
Checksum for /home/john/.local/share/.cpan/sources/authors/id/T/TO/TORBIAK/App-Git-Autofixup-0.003001.tar.gz ok
'YAML' not installed, will not store persistent state
CPAN: CPAN::Meta::Requirements loaded ok (v2.140)
CPAN: Parse::CPAN::Meta loaded ok (v2.150010)
CPAN: CPAN::Meta loaded ok (v2.150010)
CPAN: Module::CoreList loaded ok (v5.20190522)
Configuring T/TO/TORBIAK/App-Git-Autofixup-0.003001.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
WARNING: Older versions of ExtUtils::MakeMaker may errantly install README.pod as part of this distribution. It is recommended to avoid using this path in CPAN modules.
Generating a Unix-style Makefile
Writing Makefile for App::Git::Autofixup
Writing MYMETA.yml and MYMETA.json
  TORBIAK/App-Git-Autofixup-0.003001.tar.gz
  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK
Running make for T/TO/TORBIAK/App-Git-Autofixup-0.003001.tar.gz
cp lib/App/Git/Autofixup.pm blib/lib/App/Git/Autofixup.pm
cp git-autofixup blib/script/git-autofixup
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/git-autofixup
Manifying 1 pod document
Manifying 1 pod document
  TORBIAK/App-Git-Autofixup-0.003001.tar.gz
  /usr/bin/make -- OK
Running make test for TORBIAK/App-Git-Autofixup-0.003001.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/autofixup.t .. error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 253
t/autofixup.t .. 1/40
#   Failed test 'single-line change gets autofixed, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 253
t/autofixup.t .. 2/40
#   Failed test 'single-line change gets autofixed, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad36ceee0)) called at t/autofixup.t line 253

#   Failed test 'single-line change gets autofixed, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d72288)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad2d72288)) called at t/autofixup.t line 272

#   Failed test 'adjacent change gets autofixed, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d72288)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad2d72288)) called at t/autofixup.t line 272

#   Failed test 'adjacent change gets autofixed, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3ac1860)) called at t/autofixup.t line 293

#   Failed test 'adjacent change doesn't get autofixed if strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad376cc60)) called at t/autofixup.t line 303

#   Failed test 'fixups are created for additions surrounded by topic commit lines when strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 342

#   Failed test 'removed file doesn't get autofixed, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 342

#   Failed test 'removed file doesn't get autofixed, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad2d722b8)) called at t/autofixup.t line 342

#   Failed test 'removed file doesn't get autofixed, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 354

#   Failed test 're-added file doesn't get autofixed, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 354

#   Failed test 're-added file doesn't get autofixed, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be92b8)) called at t/autofixup.t line 354

#   Failed test 're-added file doesn't get autofixed, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9630)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9630)) called at t/autofixup.t line 358

#   Failed test 're-added line gets autofixed into the commit blamed for the adjacent context, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9630)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9630)) called at t/autofixup.t line 358

#   Failed test 're-added line gets autofixed into the commit blamed for the adjacent context, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9630)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9630)) called at t/autofixup.t line 358

#   Failed test 're-added line gets autofixed into the commit blamed for the adjacent context, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9450)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9450)) called at t/autofixup.t line 387

#   Failed test 'removed lines get autofixed, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9450)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9450)) called at t/autofixup.t line 387
t/autofixup.t .. 18/40
#   Failed test 'removed lines get autofixed, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9450)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be9450)) called at t/autofixup.t line 387

#   Failed test 'removed lines get autofixed, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 412

#   Failed test 'no fixups are created for upstream commits, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 412

#   Failed test 'no fixups are created for upstream commits, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 412

#   Failed test 'no fixups are created for upstream commits, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 421

#   Failed test 'fixups are created for hunks changing lines blamed by upstream if strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad376caf8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad376caf8)) called at t/autofixup.t line 445

#   Failed test 'no fixups are created for hunks changing lines blamed by upstream if strict > 0, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad376caf8)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad376caf8)) called at t/autofixup.t line 445

#   Failed test 'no fixups are created for hunks changing lines blamed by upstream if strict > 0, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 466

#   Failed test 'hunks blamed on a fixup! commit are assigned to that fixup's target, strict=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 466

#   Failed test 'hunks blamed on a fixup! commit are assigned to that fixup's target, strict=1'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 49
#       main::test_autofixup_strict(HASH(0xaaaad3bf3808)) called at t/autofixup.t line 466

#   Failed test 'hunks blamed on a fixup! commit are assigned to that fixup's target, strict=2'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3acffc0)) called at t/autofixup.t line 482

#   Failed test 'removed line gets autofixed when context=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf3670)) called at t/autofixup.t line 501

#   Failed test 'added line is ignored when context=0'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3a5d640)) called at t/autofixup.t line 510

#   Failed test 'ADJACENCY assignment is used as a fallback for multiple context targets'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3ac1890)) called at t/autofixup.t line 540

#   Failed test 'Works when run in a subdir of the repo root'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d874f8)) called at t/autofixup.t line 556

#   Failed test 'file without newline at EOF gets autofixed'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf4048)) called at t/autofixup.t line 576

#   Failed test 'multiple hunks in the same file get autofixed'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3acfff0)) called at t/autofixup.t line 613

#   Failed test 'single-line change gets autofixed when mnemonic prefixes are enabled'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be9870)) called at t/autofixup.t line 632
t/autofixup.t .. 36/40
#   Failed test 'single-line change gets autofixed when diff.external is set'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 651

#   Failed test 'exit code is 1 when some hunks are assigned'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3bf37a8)) called at t/autofixup.t line 670

#   Failed test 'multiple hunks to the same commit get autofixed'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad2d874f8)) called at t/autofixup.t line 707

#   Failed test 'only staged hunks get autofixed'
#   at t/autofixup.t line 140.
error: gpg failed to sign the data
fatal: failed to write commit object
# 32768 at t/autofixup.t line 210.
#       main::run("git", "commit", "-m", "add README") called at t/autofixup.t line 184
#       main::init_repo() called at t/autofixup.t line 101
#       eval {...} called at t/autofixup.t line 99
#       main::test_autofixup(HASH(0xaaaad3be8cb8)) called at t/autofixup.t line 735

#   Failed test 'staged hunks that aren't autofixed remain in index'
#   at t/autofixup.t line 140.
# Looks like you failed 40 tests of 40.
t/autofixup.t .. Dubious, test returned 40 (wstat 10240, 0x2800)
Failed 40/40 subtests

Test Summary Report
-------------------
t/autofixup.t (Wstat: 10240 Tests: 40 Failed: 40)
  Failed tests:  1-40
  Non-zero exit status: 40
Files=1, Tests=40,  3 wallclock secs ( 0.13 usr  0.03 sys +  0.84 cusr  1.98 csys =  2.98 CPU)
Result: FAIL
Failed 1/1 test programs. 40/40 subtests failed.
make: *** [Makefile:854: test_dynamic] Error 40
  TORBIAK/App-Git-Autofixup-0.003001.tar.gz
  /usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports TORBIAK/App-Git-Autofixup-0.003001.tar.gz
```

--- a/t/autofixup.t
+++ b/t/autofixup.t
@@ -22,6 +22,7 @@ $ENV{GIT_AUTHOR_NAME} = 'A U Thor';
 $ENV{GIT_AUTHOR_EMAIL} = 'author@example.com';
 $ENV{GIT_COMMITTER_NAME} = 'C O Mitter';
 $ENV{GIT_COMMITTER_EMAIL} = 'committer@example.com';
+$ENV{GIT_CONFIG_NOSYSTEM} = 'true';
 
 
 sub has_git {
@@ -96,6 +97,8 @@ sub test_autofixup {
     my $orig_dir = getcwd();
     my $dir = File::Temp::tempdir(CLEANUP => 1);
     chdir $dir or die "$!";
+    $ENV{HOME} = $dir;
+    $ENV{XDG_CONFIG_HOME} = "$dir/.config";
     eval {
 
         init_repo();

--- a/t/autofixup.t
+++ b/t/autofixup.t
@@ -178,7 +178,7 @@ sub test_autofixup {
 }
 
 sub init_repo {
-    run('git init');
+    run('git init --initial-branch=master');
     # git-autofixup needs a commit to exclude, since it uses the REVISION..
     # syntax. This is that commit.
     my $filename = 'README';

